doc/release-notes.md

```diff
@@ -31,15 +31,190 @@ Downgrading to a version older than v19.2.0 is not supported due to changes
+    - *(#20755)*
+
+- **`getblockfrompeer` Parameter Renaming**
+    - The named argument `block_hash` has been renamed to `blockhash` to align with the rest of the codebase.
+    - **Breaking Change:** If using named parameters, ensure to update them accordingly.
+    - *(#6149)*
+
+- **`coinjoin stop` Error Handling**
+    - `coinjoin stop` will now return an error if there is no CoinJoin mixing session to stop.
+
+- **`getcoinjoininfo` Adjustments**
+    - `getcoinjoininfo` will no longer report `keys_left` and will not incorrectly warn about keypool depletion with descriptor wallets.
+
+- **`creditOutputs` Format Change**
+    - `creditOutputs` entries in various RPCs that output transaction JSON are now shown as objects instead of strings.
+
+- **`quorum dkgsimerror` Argument Update**
+    - `quorum dkgsimerror` will no longer accept a decimal value between 0 and 1 for the `rate` argument.
+    - It will now expect an integer between **0** to **100**.
+
+- **Deprecated `protx *_hpmn` RPC Endpoints**
+    - Deprecated `protx *_hpmn` RPC entry points have been dropped in favor of `protx *_evo`.
+    - **Removed Endpoints:**
+        - `protx register_fund_hpmn`
+        - `protx register_hpmn`
+        - `protx register_prepare_hpmn`
+        - `protx update_service_hpmn`
+
+## Command-line Options
+
+### Changes in Existing Command-line Options
+
+- **`-walletnotify=<cmd>` Enhancements**
+    - Introduces new format options `%h` and `%b`.
+        - `%b`: Replaced by the hash of the block including the transaction (set to `'unconfirmed'` if the transaction is not included).
+        - `%h`: Replaced by the block height (**-1** if not included).
+
+- **`-maxuploadtarget` Format Update**
+    - Now allows human-readable byte units `[k|K|m|M|g|G|t|T]`.
+        - **Example:** `-maxuploadtarget=500g`.
+        - **Constraints:** No whitespace, `+`, `-`, or fractions allowed.
+        - **Default:** `M` if no suffix is provided.
+
+## New Functionality
+
+- **`coinjoinsalt` RPC Command**
+    - Allows manipulation of a CoinJoin salt stored in a wallet.
+        - `coinjoinsalt get`: Fetches an existing salt.
+        - `coinjoinsalt set`: Allows setting a custom salt.
+        - `coinjoinsalt generate`: Sets a random hash as the new salt.
+
+- **LLMQ Type Enhancement**
+    - **LLMQ_50_60** is enabled for **Devnet** networks.
+    - Necessary for testing on a large Devnet.
+
+## P2P and Network Changes
+
+- **Improved Onion Connectivity**
+    - Nodes with onion connectivity will attempt to maintain at least **two outbound onion connections** and will protect these connections from eviction.
+    - **Benefit:** Ensures nodes capable of accessing the onion network maintain a few onion connections, allowing network messages to propagate even if non-onion IPv4 traffic is blocked.
+    - **Security Enhancement:** Enables P2P encryption for these peers.
+    - *(#6147)*
```

Comment:
If including PR number, it should be an actual link to the PR for anyone not looking at the release notes on Github.